src/tests/ftest/util/server_utils.py

```diff
@@ -525,78 +579,93 @@ def kill(self):
 
     def clean_files(self):
         """Clean the tmpfs on the servers."""
-        scm_mount = self.runner.job.yaml_params.server_params[-1].scm_mount
-        scm_list = self.runner.job.yaml_params.server_params[-1].scm_list.value
-        clean_cmds = [
-            "find /mnt/daos -mindepth 1 -maxdepth 1 -print0 | xargs -0r rm -rf"
-        ]
-        if self.runner.job.yaml_params.is_nvme():
-            clean_cmds.append("sudo rm -rf {0};  \
-                               sudo umount {0}".format(scm_mount))
-        # scm_mount can be /mnt/daos0 or /mnt/daos1 for two daos_server
-        # instances. Presently, not supported in DAOS. The for loop needs
-        # to be updated in future to handle it. Single instance pmem
-        # device should work now.
-        if self.runner.job.yaml_params.is_scm():
-            for value in scm_list:
-                clean_cmds.append("sudo umount {}; \
-                                   sudo wipefs -a {}"
-                                  .format(scm_mount, value))
-        self.log.info("Cleanup of %s directory.", str(scm_mount))
-        pcmd(self._hosts, "; ".join(clean_cmds), False)
-
+        is_dcpm = self.runner.job.yaml_params.is_scm()
+        is_nvme = self.runner.job.yaml_params.is_nvme()
+        commands = []
+        for server_params in self.runner.job.yaml_params.server_params:
+            scm_list = server_params.scm_list.value
+            scm_mount = server_params.scm_mount.value
+            scm_size = server_params.scm_size.value
+
+            self.log.info("Cleaning up the %s directory.", str(scm_mount))
+
+            # Remove any files left over on each mount point
+            if is_nvme:
+                commands.append("sudo rm -fr {}".format(scm_mount))
+            else:
+                # Also ensure the tmpfs/dcpm mount point exists
+                commands.extend([
+                    "if [ ! -e {} ]".format(scm_mount),
+                    "then sudo mkdir {}".format(scm_mount),
+                    "else find {} -mindepth 1 -maxdepth 1 -print0 | "
+                    "xargs -0r rm -rf".format(scm_mount),
+                    "fi"])
+
+            # Unmount the nvme drive
+            if is_nvme:
+                commands.append("sudo umount {}".format(scm_mount))
+
+            # Clean the dcpm device
+            if is_dcpm:
+                for dcpm_mount in scm_list:
+                    commands.extend([
+                        "sudu umount {}".format(dcpm_mount),
+                        "sudo wipefs -a {}".format(dcpm_mount)])
+
+            # Ensure the tmpfs mount point is mounted
+            if not is_dcpm and not is_nvme:
+                commands.append(
+                    "sudo mount -t tmpfs -o size={}G tmpfs {}".format(
+                        scm_size, scm_mount))
```

Comment:
Likewise here, ideally we should be testing that the control plane is able to do this stuff correctly.